src/addMeToYourFriends.ts

```diff
@@ -0,0 +1,146 @@
+import { html, TemplateResult } from "lit-html";
+import { styleMap } from "lit-html/directives/style-map";
+import { DataBrowserContext, LiveStore } from "pane-registry";
+import * as UI from "solid-ui";
+import { padding, textCenter } from "./baseStyles";
+
+let buttonContainer = <HTMLDivElement>document.createElement("div");
+//TODO create dedicated component in UI
+const positiveFrontendMessageDiv = <HTMLDivElement>document.createElement("div");
+positiveFrontendMessageDiv.setAttribute(
+  "style",
+  "margin: 0.1em; padding: 0.5em; border: 0.05em solid gray; background-color: #efe; color:black;"
+);
+
+const styles = {
+  button: styleMap({ ...textCenter(), ...padding() }),
+};
+
+//Same not logged in error message like on 'Chat with me' buton
+const userNotLoggedInErrorMessage = "Current user not found! Not logged in?";
+const internalErrorMessage = "An internal error occured!";
+const friendWasAddedSuccesMessage = "Friend was added!";
+const friendExistsMessage = "Friend already exists";
+const addMeToYourFriendsButtonText = "Add me to your friend list";
+const friendExistsAlreadyButtonText = "Already part of friend list";
+
+const addMeToYourFriendsHtml = (
+  subject: UI.rdf.NamedNode,
+  context: DataBrowserContext
+): TemplateResult => {
+  buttonContainer = context.dom.createElement("div");
+  const button = createAddMeToYourFriendsButton(subject, context);
+  buttonContainer.appendChild(button);
+  return html` <div style="${styles.button}">${buttonContainer}</div> `;
+};
+
+const createAddMeToYourFriendsButton = (
+  subject: UI.rdf.NamedNode,
+  context: DataBrowserContext
+): HTMLButtonElement => {
+  const button = UI.widgets.button(
+    context.dom,
+    undefined,
+    addMeToYourFriendsButtonText,
+    undefined,
+    {
+      needsBorder: true,
+    }
+  );
+  //this is to make clear which style we have, for code readability
+  button.setAttribute("class", "textButton-0-1-3"); //style of 'Primary' UI button with needsBorder=true
+
+  const me = UI.authn.currentUser();
+  const store = context.session.store;
+
+  if (checkIfAnyUserLoggedIn(me)) {
+    checkIfFriendExists(store, me, subject).then((friendExists) => {
+      if (friendExists) {
+        button.innerHTML = friendExistsAlreadyButtonText.toUpperCase();
+        button.setAttribute("class", "textButton-0-1-3");
+      } else button.setAttribute("class", "textButton-0-1-2"); //style of 'Primary' UI button with needsBorder=false
+    });
+  }
+
+  button.addEventListener(
+    "click",
+    () =>
+      saveNewFriend(subject, context)
+        .then(() => reportToFrontend(true, context, friendWasAddedSuccesMessage))

```

Comment:
suggestion: I think I've seen notify before versus reportTo. perhaps have a quick look and if so could potentially change just for consistency.  I'm not 100% about this.  just a thought.

Reply:
I only found announce and mention and I renamed my functions to these.